### PR TITLE
Toggleable Hardsuit Helmets

### PR DESCRIPTION
**Pull request by ElectroJr**
Makes hardsuit helmets part of the hardsuit itself, which can be toggled on or off using an action. This just works by adding a `ContainerSlot` to the hardsuits that stores a helmet, which is equipped whenever the action is toggled.

The main new additions are `ToggleableClothingSystem`, `ToggleableClothingComponent`, and `AttachedClothingComponent`, but there are some other changes around equipment sounds, action icons, and blocking hardsuit helmets from being moved to a users hands (some of which is a bit janky).

The actual action icon is very similar to the toggle flashlight icon, which has a little torch in the corner. I'm not sure how best to fix that. I guess the icon could just show the hardsuit instead of the helmet?

These changes also mean that most helmets shouldn't spawn in on their own anywhere. If some of the maps have them mapped in, then they need to be removed.

:cl:
- tweak: Hardsuit helmets are now part of a hardsuit. They can be toggled via an action.

### CODE REVIEW

**Comment by Emisse** [comment]
i dont think we have any hardsuit helmets mapped in, unless this applies to eva

**Comment by ElectroJr** [comment]
> unless this applies to eva

currently this does include eva suits, but I could just make the toggle stuff specific to hard suits

**Comment by Emisse** [comment]
i think that would be best imo

**Comment by ZeroDayDaemon** [comment]
finally

File: Content.Shared/Inventory/InventorySystem.Equip.cs [hunk 1]
```diff
@@ -340,6 +359,24 @@ public bool CanEquip(EntityUid actor, EntityUid target, EntityUid itemUid, strin
...
+            {
+                filter = Filter.Pvs(target);
+
+                // don't play double audio for predicted interactions
+                if (predicted)
+                    filter.RemoveWhereAttachedEntity(entity => entity == actor);
+            }
+
+            SoundSystem.Play(filter, item.UnequipSound.GetSound(), target, AudioParams.Default.WithVolume(-2f));
+        }
```
Comment (mirrorcult) [inline Content.Shared/Inventory/InventorySystem.Equip.cs:378]: wow this is horrific boilerplate, not going to force you to change that here though

File: Content.Shared/Item/SharedItemComponent.cs [hunk 2]
```diff
@@ -37,6 +37,20 @@ public int Size
...
+        /// <remarks>
+        ///     This should almost always be true for items. But in some special cases, an item can be equipped but not
+        ///     picked up. E.g., hardsuit helmets are attached to the suit, so we want to disable things like the pickup
+        ///     verb.
+        /// </remarks>
+        [DataField("canPickup")]
+        public bool CanPickup = true;
+
+        [DataField("quickEquip")]
+        public bool QuickEquip = true;
```
Comment (mirrorcult) [inline Content.Shared/Item/SharedItemComponent.cs:52]: shouldnt this be on clothing

Comment (mirrorcult) [reply Content.Shared/Item/SharedItemComponent.cs:52]: well i guess item and clothing are already insaenly fucked up and coupled so if it has to be like this thats okay

**Review by mirrorcult — CHANGES_REQUESTED** [review]
need to ping timrod/peptide/emisse in cartography when this is merged

**Review by mirrorcult — APPROVED** [review]
hit the wrogn one

File: Content.Shared/Item/SharedItemComponent.cs (hunk 2 above)
Comment (ElectroJr) [reply Content.Shared/Item/SharedItemComponent.cs:52]: It does also function for quickly moving items into pockets, so IMO its probably fine.

File: Content.Shared/Inventory/InventorySystem.Equip.cs (hunk 1 above)
Comment (ElectroJr) [reply Content.Shared/Inventory/InventorySystem.Equip.cs:378]: yeah it is. there is currently no nice way to handle audio prediction. Usually there are separate client/server systems but here I just combined them with `_netMan.IsClient` which also makes it all uglier.

**Pull request merged**